src/adagrams.js

```diff
@@ -1,15 +1,155 @@
+const LETTER_POOL_COUNT = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+const SCORE_CHART = {
+  A: 1,
+  E: 1,
+  I: 1,
+  O: 1,
+  U: 1,
+  L: 1,
+  N: 1,
+  R: 1,
+  S: 1,
+  T: 1,
+  D: 2,
+  G: 2,
+  B: 3,
+  C: 3,
+  M: 3,
+  P: 3,
+  F: 4,
+  H: 4,
+  V: 4,
+  W: 4,
+  Y: 4,
+  K: 5,
+  J: 8,
+  X: 8,
+  Q: 10,
+  Z: 10,
+};
+// --------------------------------------------------------wave 1------------------------------------------------
 export const drawLetters = () => {
   // Implement this method for wave 1
+  let count_obj = {};
+  let letter_list = [];
+  let new_letter = "";
+  const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  for (let i = 0; i < 10; i++) {
+    new_letter = characters.charAt(
+      Math.floor(Math.random() * characters.length)
+    );
+    if (
+      new_letter in count_obj &&
+      LETTER_POOL_COUNT[new_letter] > count_obj[new_letter]
+    ) {
+      count_obj[new_letter] += 1;
+      letter_list.push(new_letter);
+    } else if (!(new_letter in count_obj)) {
+      count_obj[new_letter] = 1;
+      letter_list.push(new_letter);
+    } else {
+      continue;
+    }
```

Comment:
This else and continue are not necessary. The code will automatically continue at the end of the if statement. You do not need to tell it to continue.